__tests__/AdminLogin.js

```diff
@@ -137,7 +138,8 @@ describe('AdminLogin', () => {
 
   it('redirects to homepage if the user has been previously logged in', async () => {
     getMockCall('/v4/user/', userResponse);
-    getMockCallTimes('/v4/info/', AWSInfoResponse, 2);
+    getInfo.mockResolvedValueOnce(AWSInfoResponse);
+    getInfo.mockResolvedValueOnce(AWSInfoResponse);
```

Comment:
My thinking is that tests is not the place to figure out how many calls we are doing, we should keep track of this while we are developing, or on performance audits.

I am ok if you guys want to do it here, it's just that I would love to forget about this in tests and make them lighter.

Reply:
I am with you on that, it makes testing more annoying than it should be.

However, here I just try to make a drop-in replacement. We should definitely discuss removing this pattern, but I believe the change should happen in another PR

Reply:
Agree :-)

Reply:
I think there are some benefits to having a strict watch on the request behaviour... We can catch regressions or issues that we wouldn't otherwise if we would just set up a set of persistent requests or mocks of these new functions without checking how often they are called.

That said I'm happy to discuss it with everyone. 

Also I think the integration tests are getting close to complete? We shouldn't be adding much more integration tests after this set. 

There are some refactorings to do to fix a bit of flappyness, but I think the focus should start shifting to testing the components individually.

I did the Version Picker component in a kind of test driven way, and it was pretty nice to do it like that. And in those more focused contexts, there is no need to worry about mocking or nocking any requests at all.

Reply:
Let's discuss it on Friday 🚀